pkgs/by-name/op/opencloud-oidc-webfinger-proxy/package.nix

```
@@ -0,0 +1,38 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  makeWrapper,
+}:
+
+buildGoModule rec {
+  pname = "opencloud-oidc-webfinger-proxy";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "2bros-group";
+    repo = "opencloud-oidc-webfinger-proxy";
+    rev = "v${version}";
+    hash = "sha256-dI8rm/7Xly0cDjRYxKv6l6b2003YMcwz0k77KxBu/AU=";
+  };
+
+  # Generate a minimal go.mod before building because upstream does not provide one.
+  preBuild = ''
+    if [ ! -f go.mod ]; then
+      echo "module github.com/2bros-group/opencloud-oidc-webfinger-proxy" > go.mod
+      go mod tidy
+    fi
+  '';
+
+  vendorHash = null;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  meta = {
+    description = "A lightweight Go-based reverse proxy that dynamically rewrites href fields in WebFinger responses for OpenID Connect (OIDC) clients.";
```

```suggestion
    description = "Lightweight Go-based reverse proxy that dynamically rewrites href fields in WebFinger responses for OpenID Connect (OIDC) clients";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes